### ISSUE TITLE
Configure datadog service for postgres

### ISSUE THREAD

**Issue by acozine** [pulibrary/princeton_ansible#4357]
We are not seeing postgres 15 logs in the Datadog UI. There's a warning on the host entry for lib-postgres-prod1:
```
Instance #initialization[ERROR]: {"Core Check Loader":"Check postgres not found in Catalog","JMX Check Loader":"check is not a jmx check, or unable to determine if it's so","Python Check Loader":"unable to import module 'postgres': No module named postgres"}
```

On the lib-postgres-prod1 VM:
- the datadog agent is installed
- we've set `log_enabled: true` in the `/etc/datadog-agent/datadog.yml` file
- a `conf.yaml` file exists in `/etc/datadog-agent/conf.d/postgres.d`
- the psql logs are in JSON format and stored in `/var/lib/postgresql/15/main/pg_log/`
- log filename pattern is `'postgresql-%Y-%m-%d_%H%M%S.json'`

We may need to update this to match the recommended Datadog log settings - to find instructions, go to [this docs page](https://app.datadoghq.com/logs/onboarding/server) and select Postgresql.

**Comment by acozine** [comment]
Dug into this a bit more. We found one or two things that might be causing or contributing to this issue:
- the postgres check configuration was put into alphabetized YAML, which may mean the keys are wrong - the file originally looked like this:
```
init_config: null
instances:
-   dbm: true
    host: localhost
    password: (scrubbed)
    port: 5432
    username: datadog
logs:
-   path: /var/lib/postgresql/15/main/pg_log/*.json
    service: database
    source: postgresql
    sourcecategory: postgresql_logs
    tags: postgresql, env:prod, role:postgresql
    type: file
```
and we tried updating it (artisanally, on the server) to look like this:
```
init_config: null
instances:
  - host: localhost
    dbname: postgres
    # dbm: true
    host: localhost
    password: (scrubbed)
    port: 5432
    username: datadog
logs:
  - type: file
    path: /var/lib/postgresql/15/main/pg_log/postgresql-%Y-%m-%d_%H%M%S.json
    service: database
    source: postgresql
    sourcecategory: postgresql_logs
    tags: postgresql, env:prod, role:postgresql
    # type: file
```
- the log files have 0600 permissions and are owned by the postgres user, so the datadog-agent user cannot read them - we did not have time to try changing this

**Comment by acozine** [comment]
Today we made more progress!
- We added a log_processing_rule to datadog config so we no longer send `LOG` level messages to Datadog.
- We modified the log pre-processing pipeline for JSON logs to flag the severity of log entries from postgres according to postgres' own `error_severity` JSON key (see [this doc page](https://docs.datadoghq.com/logs/guide/logs-show-info-status-for-warnings-or-errors/#json-logs)).
